src/components/shared/Popover.js

```diff
@@ -65,14 +75,21 @@ class Popover extends Component {
 
   renderPopover() {
     const { onMouseLeave } = this.props;
-    const { top, left, dir } = this.state;
+    const { top, left, dir, targetMid } = this.state;
+
+    const arrow = new BracketArrow({
+      dir: dir === "up" ? "down" : "up",
```

Comment:
So here we also want to so something similar to the comment that i made up on `getPopoverCoords()` -- maybe give this a stab on your own and see how this can be refactored for readability?